### ISSUE TITLE
Support for Chrome

### ISSUE THREAD

**Issue by zackmark29** [rowrawer/stream-detector#9]
This is really great. I just found it today
Could you please support also for chrome extension?

**Comment by rowrawer** [comment]
I'll look into it, but no promises.

**Comment by xplorr** [comment]
A Chrome/Chromium version would be welcome!

**Comment by rowrawer** [comment]
I have attached a crx file to the [v2.6.4 release](https://github.com/rowrawer/stream-detector/releases/tag/v2.6.4). Let me know if it works if you're interested.

**Comment by xplorr** [comment]
crx file cannot be enabled in Chrome because it was not found in the extension store

**Comment by rowrawer** [comment]
Yeah, you have to extract it and load it as an unpacked extension I think.

**Comment by xplorr** [comment]
Unpacked and loaded gives error:
found unexpected key 'browser_style' __MSG_extText__
```
{
	"description": "__MSG_extText__",
	"manifest_version": 2,
	"name": "The Stream Detector",
	"default_locale": "en",
	"version": "2.6.4",
	"permissions": [
		"<all_urls>",
		"tabs",
		"clipboardWrite",
		"notifications",
		"storage",
		"webRequest"
	],
	"background": {
		"scripts": ["background.js"]
	},
	"icons": {
		"16": "img/icon-dark-16.png",
		"48": "img/icon-dark-48.png",
		"96": "img/icon-dark-96.png"
	},
	"options_ui": {
		"browser_style": true,
		"open_in_tab": false,
		"page": "content/options.html"
	},
	"browser_action": {
		"browser_style": true,
		"default_title": "The Stream Detector",
		"default_popup": "content/popup.html",
		"default_icon": {
			"16": "img/icon-dark-16.png",
			"48": "img/icon-dark-48.png"
		},
		"theme_icons": [
			{
				"dark": "img/icon-dark-96.png",
				"light": "img/icon-light-96.png",
				"size": 96
			},
			{
				"dark": "img/icon-dark-48.png",
				"light": "img/icon-light-48.png",
				"size": 48
			},
			{
				"dark": "img/icon-dark-16.png",
				"light": "img/icon-light-16.png",
				"size": 16
			}
		]
	}
}
```

**Comment by xplorr** [comment]
But seems to work ok. Thx!

**Comment by rowrawer** [comment]
As long as it works then it's all good. I've removed these lines from the file anyway as they're not currently very useful. Thanks for testing.

**Comment by xplorr** [comment]
Why not release it in the Chrome store?
It may be very useful for Chrome users.

**Comment by rowrawer** [comment]
It costs money to publish stuff in the Chrome store and I'm not about that life.

**Comment by xplorr** [comment]
Thanks a lot for this Chrome extension!

**Comment by Bibbbi** [comment]
I found a bug in the Chrome extension the_stream_detector-2.7.4.crx

Uncaught ReferenceError: browser is not defined: background.js:212

**Find:**
browser.commands.onCommand.addListener(cmd => {
**Replace:**
chrome.commands.onCommand.addListener(cmd => {

[https://i.imgur.com/laxD7eO.png](url)

**Comment by rowrawer** [comment]
My bad. I've updated the crx file in the releases tab, should be fine now.

**Comment by goooglefc** [comment]
The chrome extension doesn't pickup the correct Referer for embedded M3U streams.

I know you don't provide support for Chrome, but, if possible, please look into this.

**Comment by rowrawer** [comment]
I doubt I will, but if I did, you'd have to be more descriptive. Which website, what is the referer, what should it be, is it different in Firefox, etc.

**Comment by rowrawer** [comment]
The issue can be kept open for the time being.

**Comment by goooglefc** [comment]
As an example you can look at https://sportsonline.to/channels/pt/sporttv3.php
The actual referer for this stream is "https://sastreaming.net" but the extension just picks up the current URL as referer.
There are many more like this.
The firefox extension obviously works fine on links like these and displays the correct referer.

**Comment by rowrawer** [comment]
I looked into it and it was a quick fix. Should be working fine in v2.8.7.

**Comment by goooglefc** [comment]
So far so good.
Thanks you.

**Comment by pc00per** [comment]
> It costs money to publish stuff in the Chrome store and I'm not about that life.

I thought it's free to host extensions on web store. Is that not true?

**Comment by rowrawer** [comment]
> I thought it's free to host extensions on web store. Is that not true?

You don't have to pay per extension, but there is a one time fee. It's cheap, and I might consider doing it, but I'm a bit wary about having to support two different browsers simultaneously.

**Comment by pc00per** [comment]
> but I'm a bit wary about having to support two different browsers simultaneously.

I really appreciate you're atleast building .crx for chromium browsers too, which the support work is already done for us mate. Hosting on web store is just the last 10% task imho.
If you just need any help hosting & maintaining on web store, we could help you through donations, or find some maintainer for you...

**Comment by rowrawer** [comment]
> If you just need any help hosting & maintaining on web store, we could help you through donations, or find some maintainer for you...

My reservations stem from the fact that this is a personal project which I've been maintaining for my own use, and outsourcing it to maintainers or getting donations would make this addon a responsibility and, inevitably, a chore. If I didn't mind that I'd have uploaded it to the Web Store a long time ago.

**Comment by nammminn2** [comment]
https://share.cleanshot.com/VEnEf9 i got this error when i tried to load it to chrome

**Comment by Metamoran** [comment]
> https://share.cleanshot.com/VEnEf9 i got this error when i tried to load it to chrome

Did you try to install it by clicking the CRX link in the releases page? If that's the case, right click and save the CRX file, open "chrome://extensions" in a new tab, then drag&drop said file in it instead. Worked fine for me.

PS: I just discovered this extension, and it works wonders in both Firefox and Chrome. Catches quite a few media streams other extensions don't. Thanks a lot, rowrawer.

**Comment by drdedus** [comment]
hello
is out there something similar for Android?
thank you
